zookeeper-server/src/main/java/org/apache/zookeeper/server/ByteBufferRequestRecord.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *uuuuu
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "/RequuuAS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.server;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.jute.Record;
+
+public class ByteBufferRequestRecord implements RequestRecord {
+
+    private final ByteBuffer request;
+
+    private volatile Record record;
+
+    public ByteBufferRequestRecord(ByteBuffer request) {
+        this.request = request;
+    }
+
+    @Override
+    public <T extends Record> T readRecord(Class<T> clazz) throws IOException {
+        if (record != null) {
+            try {
+                return clazz.cast(record);
+            } catch (Exception e) {
+                throw new IOException(e);
+            }
+        }
+
+        try {
+            record = clazz.getDeclaredConstructor().newInstance();
```

Comment:
we shouldn't use reflection here in a hot path (while processing requests) reflection is really slow

Reply:
Make sense. I think a `Supplier<T>` as `T::new` can help here.

Reply:
resolved at 03b16e330ddd4c0bee2d8b1591ca74b2a35748af